plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatController.kt

```diff
@@ -0,0 +1,649 @@
+import software.aws.toolkits.core.utils.debug
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.warn
+import software.aws.toolkits.jetbrains.core.coroutines.EDT
+import software.aws.toolkits.jetbrains.core.gettingstarted.requestCredentialsForQ
+import software.aws.toolkits.jetbrains.core.webview.BrowserState
+import software.aws.toolkits.jetbrains.services.amazonq.QWebviewPanel
+import software.aws.toolkits.jetbrains.services.amazonq.auth.AuthController
+import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AMAZON_Q_WINDOW_ID
+import software.aws.toolkits.jetbrains.services.codewhisperer.customization.CodeWhispererModelConfigurator
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.programmingLanguage
+import software.aws.toolkits.jetbrains.services.cwc.clients.chat.model.ChatRequestData
+import software.aws.toolkits.jetbrains.services.cwc.clients.chat.model.TriggerType
+import software.aws.toolkits.jetbrains.services.cwc.controller.chat.messenger.ChatPromptHandler
+import software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper
+import software.aws.toolkits.jetbrains.services.cwc.controller.chat.userIntent.UserIntentRecognizer
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.ActiveFileContextExtractor
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.ExtractionTriggerType
+import software.aws.toolkits.jetbrains.services.cwc.inline.listeners.InlineChatFileListener
+import software.aws.toolkits.jetbrains.services.cwc.messages.ChatMessage
+import software.aws.toolkits.jetbrains.services.cwc.messages.ChatMessageType
+import software.aws.toolkits.jetbrains.services.cwc.storage.ChatSessionStorage
+import software.aws.toolkits.telemetry.FeatureId
+import java.util.Stack
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicBoolean
+
+@Service(Service.Level.PROJECT)
+class InlineChatController(
+    private val project: Project,
+    private val scope: CoroutineScope,
+) : Disposable {
+    private var currentPopup: JBPopup? = null
+    private var rangeHighlighter: RangeHighlighter? = null
+    private val partialUndoActions = Stack<() -> Unit>()
+    private val partialAcceptActions = Stack<() -> Unit>()
+    private var insertionLine = AtomicInteger(-1)
+    private val sessionStorage = ChatSessionStorage()
+    private val telemetryHelper = TelemetryHelper(project, sessionStorage)
+    private val shouldShowActions = AtomicBoolean(false)
+    private val isInProgress = AtomicBoolean(false)
+    private var metrics: InlineChatMetrics? = null
+    private var canPopupAbort = AtomicBoolean(true)
+    private var currentSelectionRange: RangeMarker? = null
+
+    init {
+        InlineChatFileListener(project, this).apply {
+            project.messageBus.connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, this)
```

Comment:
message bus connections need disposable 
```suggestion
            project.messageBus.connect(this).subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, this)
```